### ISSUE TITLE
Demonstration of using GridSearchCV and RandomSearchCV

### ISSUE THREAD

**Issue by adam2392** [adam2392/mne-hfo#14]
#### Describe the problem
Demonstrate how using the scikit-learn's api to find optimal combination of hyperparameters for a Detector.
Since optimal combination depends on the Detector having a ground-truth to compare to, then this must be 
run on the Fedele dataset.

This should be summarized inside the tutorial notebook.

#### Describe your solution
This would be a two-step PR:

### Refactoring match_detections to be simple API and return y_true and y_pred as lists

1. Refactor `match_detections`
Match detections does not need `bn` kwarg. Seems unnecessary considering we are assuming a structured DataFrame. 
It also does not need `freq_name`. Instead it should look like this:

```Python
def match_detections(ytrue_df, ypredict_df, label:str=None, sec_margin: float=1.):
```

where `label` if passed in will filter which label to match (e.g. `fastripple`, or `ripple`, or just plain old `hfo`). One 
should be able to infer the sampling rate based on the DataFrame. In addition, rn the function is unusable and hard to read, so if you want to keep it, keep it, but make sure you can describe how it's working. FYI, I copied it over from epycom, so no guarantee is functions in any way (no tests).

2. Add unit test for match_detections

See existing unit tests and how it's done.

### Implement Detector.predict() to make *SearchCV functions work and demonstrate

Next, make the scikit-learn's *SearchCV functions work, which will require implementing `predict` and `score`. 
I believe `predict` is already done. It just needs to "detect the HFOs". `score` needs to be implemented. Score 
should essentially call `match_detections` and then compute a metric using `scoring_func`. It should allow 
kwargs that need to be passed to `match_detections` (e.g. label, or sec_margin to customize how we want to score). 

1. implement `score` function 
2. demonstrate now in sequence using the GridSearchCV and RandomSearchCV functions on one subject of the Fedele dataset.
